onnxruntime/core/optimizer/reshape_fusion.cc

```diff
@@ -344,7 +352,8 @@ bool ReshapeFusion::Fuse_Subgraph(Node& reshape, Graph& graph, const logging::Lo
   }
   const Node& concat = *p_concat;
 
-  if (!graph_utils::IsSupportedOptypeVersionAndDomain(concat, "Concat", {1, 4, 11, 13})) {
+  if (!graph_utils::IsSupportedOptypeVersionAndDomain(concat, "Concat", {1, 4, 11, 13}) &&
+      !graph_utils::IsSupportedOptypeVersionAndDomain(concat, "ConcatTraining", {1}, kMSDomain)) {
```

Comment:
For ConcatTraining, do we need check whether the 2nd output is still needed by other consumers?

Reply:
In line:361: 
if (!optimizer_utils::CheckOutputEdges(graph, concat, 1)) { return false;}
this will return false if ConcatTraining have 2nd output.